.github/workflows/tests.yaml

```diff
@@ -10,14 +10,14 @@ env:
 jobs:
   tests:
     runs-on: ubuntu-latest
-    name: Run lint and tests tests
+    name: Run lint and tests
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-    - name: Checkout main
-      uses: actions/checkout@v2
+    - name: Checkout main, so we have it to refer to
+    - uses: actions/checkout@v2
       with:
           ref: main
+    - name: Checkout PR branch
+      uses: actions/checkout@v2
```

Comment:
Does the second checkout here now overwrite the first one?

Reply:
Yes, that was the problem originally. But we need a reference to main, as the update.sh script that we test does 

`git checkout -f main`

And if we don't check it out first, that will fail.

Reply:
So, we do need a checkout of it, but we need the actual branch to be checkout last

Reply:
FTR, we copy the current checkout into docker container and test stuff in their, the git repo in the github runner is never actually updated back to main

Reply:
Both checkouts do a `fetch` and `checkout`, so the 2nd checkout *does* overwrite the first (which is why the order is important). But the first checkout has done a `git fetch main`, so that ref exists in the local repo and can be checked out in tests later on.